console/src/components/Services/Data/TableModify/RootFields.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import ExpandableEditor from '../../../Common/Layout/ExpandableEditor/Editor';
+import RootFieldEditor from '../Common/ReusableComponents/RootFieldEditor';
+import { modifyRootFields, setCustomRootFields } from './ModifyActions';
+
+const RootFields = ({
+  existingRootFields,
+  rootFieldsEdit,
+  dispatch,
+  tableName,
+}) => {
+  const setRootFieldsBulk = rf => {
+    dispatch(modifyRootFields(rf));
+  };
+
+  const onChange = (field, alias) => {
+    const newRootFields = {
+      ...rootFieldsEdit,
+      [field]: alias,
+    };
+    dispatch(modifyRootFields(newRootFields));
+  };
+
+  const note =
+    'You can customize the GraphQL root fields associated with this table.';
+
+  const rootFields = (aliases, disabled) => {
+    return (
```

Comment:
Lets change this behaviour for disabled as currently onChange functions will be sent while component is disabled. Or is it required? if not 

we can have a JSON object 
```
const onChangeHandlers = {};
if (!disabled) {
    // Attach handlers to specific keys
} 

// Rest

<RootFieldEditor
   ... rest
   { ...onChangeHandlers}
/>
```